Cinema system/Cinema system/Services/SeanceService.cs

```diff
@@ -0,0 +1,269 @@
+                return false;
+            }
+
+            return true;
+        }
+
+        public bool СheckForNonmatchingSeanceTime(int seanceId, SeanceInfo seance)
+        {
+            var prevSeance = _db.Seances
+                .Where(s => s.HallId == seance.HallId
+                   && seance.ShowDate.Date == s.ShowDate.Date
+                   && s.ShowDate < seance.ShowDate
+                   && s.Id != seanceId)
+                .OrderBy(s => s.ShowDate)
+                .LastOrDefault();
+
+            if (prevSeance == null)
+            {
+                return true;
+            }
+
+            if (seance.ShowDate < prevSeance.ShowDate.Add(prevSeance.ShowTime))
+            {
+                return false;
+            }
+
+            var nextSeance = _db.Seances
+                .Where(s => s.HallId == seance.HallId
+                   && seance.ShowDate.Date == s.ShowDate.Date
+                   && s.ShowDate > seance.ShowDate)
+                .OrderBy(s => s.ShowDate)
+                .FirstOrDefault();
+
+            if (nextSeance == null)
+            {
+                return true;
+            }
+
+            if (nextSeance.ShowDate < seance.ShowDate.Add(seance.ShowTime))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        public int CreateSeance(SeanceInfo seance)
+        {
+            Seance newSeance = new Seance
+            {
+                MovieId = seance.MovieId,
+                HallId = seance.HallId,
+                ShowDate = seance.ShowDate,
+                ShowTime = seance.ShowTime
+            };
+
+            foreach(SeanceSeatView seanceSeat in seance.SessionSeats)
+            {
+                var newSeanceSeat = new SeanceSeat
+                {
+                    SeanceId = newSeance.Id,
+                    SeatId = seanceSeat.SeatId,
+                    Price = seanceSeat.Price,
+                    IsAvailable = seanceSeat.IsAvailable
+                };
+
+                _db.SeanceSeats.Add(newSeanceSeat);
+            }
+
+            _db.Seances.Add(newSeance);
+
+            _db.SaveChanges();
+
+            return newSeance.Id;
+        }
+
+        public void EditSeance(int seanceId, SeanceInfo seance)
+        {
+            Seance editSeance = _db.Seances.Find(seanceId);
+
+            editSeance.ShowDate = seance.ShowDate;
+            editSeance.ShowTime = seance.ShowTime;
+            editSeance.MovieId = seance.MovieId;
+            editSeance.HallId = seance.HallId;
+            editSeance.ShowTime = seance.ShowTime;
+
+            foreach (SeanceSeatView seanceSeat in seance.SessionSeats)
+            {
+                var editSeanceSeat = _db.SeanceSeats
+                    .SingleOrDefault(s => (s.SeanceId == seanceId && s.SeatId == seanceSeat.SeatId));
+
+                editSeanceSeat.Price = seanceSeat.Price;
+                editSeanceSeat.IsAvailable = seanceSeat.IsAvailable;
+            }
```

Comment:
Can you see a problem here? How can it be avoided?

Reply:
Problem with changing isAvailable field?

Reply:
No, the problem is with database requests.